src/Generator/CreditMemoPdfFileGenerator.php

```diff
@@ -74,21 +53,41 @@ public function generate(string $creditMemoId): CreditMemoPdf
 
         $filename = str_replace('/', '_', $number) . self::FILE_EXTENSION;
 
-        $pdf = $this->pdfGenerator->getOutputFromHtml(
-            $this->twig->render($this->template, [
-                'creditMemo' => $creditMemo,
-                'creditMemoLogoPath' => $this->fileLocator->locate($this->creditMemoLogoPath),
-            ]),
-            $this->pdfOptionsGenerator ? $this->pdfOptionsGenerator->generate() : []
-        );
+        $templateParams = [
+            'creditMemo' => $creditMemo,
+            'creditMemoLogoPath' => $this->fileLocator->locate($this->creditMemoLogoPath),
+        ];
+
+        if (null !== $this->twigToPdfGenerator) {
+            $pdf = $this->twigToPdfGenerator->generate($this->template, $templateParams);
+        } elseif (null !== $this->pdfGenerator && null !== $this->twig) {
+            $pdf = $this->pdfGenerator->getOutputFromHtml(
+                $this->twig->render($this->template, $templateParams),
+                $this->pdfOptionsGenerator ? $this->pdfOptionsGenerator->generate() : []
+            );
+        } else {
+            throw new \LogicException(sprintf('You must pass at least $twigToPdfGenerator to %s constructor.', self::class));
+        }
 
         return new CreditMemoPdf($filename, $pdf);
     }
 
     private function checkDeprecations(): void
     {
-        if (null === $this->pdfOptionsGenerator) {
-            @trigger_error(sprintf('Not passing a $pdfOptionsGenerator to %s constructor is deprecated since sylius/refund-plugin 1.1 and will be prohibited in 2.0.', self::class), \E_USER_DEPRECATED);
+        if (null !== $this->twig) {
+            @trigger_error(sprintf('Passing %s as the second argument to %s constructor is deprecated since sylius/refund-plugin 1.1 and will be prohibited in 2.0.', Environment::class, self::class), \E_USER_DEPRECATED);
```

Comment:
```suggestion
            @trigger_error(sprintf('Passing %s as the second argument to %s constructor is deprecated since sylius/refund-plugin 1.2 and will be prohibited in 2.0.', Environment::class, self::class), \E_USER_DEPRECATED);
```
I've changed the base branch of this PR, as this refactor with deprecations should appear in the newest minor, not patch version